core/point/point.go

```diff
@@ -17,7 +29,38 @@ func New(x, y int64) *Point {
 }
 
 // X returns the x-value.
-func (p *Point) X() int64 { return p.x }
+func (pt *Point) X() int64 { return pt.x }
 
 // Y returns the y-value.
-func (p *Point) Y() int64 { return p.y }
+func (pt *Point) Y() int64 { return pt.y }
+
+// ToSGF converts a pointer-type (immutable) *Point
+// to an SGF Point (two letter string). The returned value is 0-indexed.
+func (pt *Point) ToSGF() string {
+	sgfOut := ""
+	if (pt != nil) && (pt.X() <= 51) && (pt.Y() <= 51) {
+		sgfX := string(PointSGFSlce[pt.X()])
+		sgfY := string(PointSGFSlce[pt.Y()])
+		sgfOut = sgfX + sgfY
+	} else {
+		panic("Error: *Point entries must not be nil for either" +
```

Comment:
Don't panic -- just return an error as a value. Panic is for exceptional cases and programming errors -- stuff that should cause the program to crash.

- https://blog.golang.org/error-handling-and-go